test/t_web/channels/feed_channel_test.exs

```diff
@@ -72,17 +72,18 @@ defmodule TWeb.FeedChannelTest do
                  "id" => m3.id,
                  "profile" => %{name: "mate-3", story: [], user_id: p3.id, gender: "M"},
                  "timeslot" => %{"selected_slot" => s2},
-                 "audio_only" => false
+                 "audio_only" => false,
+                 "expiration_date" => ~U[2021-10-07 12:16:07Z]
                },
                %{
                  "id" => m2.id,
                  "profile" => %{name: "mate-2", story: [], user_id: p2.id, gender: "N"},
-                 "contact" => %{
-                   "contacts" => %{"whatsapp" => "+79666666666"},
-                   "picker" => p2.id,
-                   "opened_contact_type" => nil
+                 "timeslot" => %{
```

Comment:
> # add contact to m2, but it won't be returned since there is a timeslot as well
(from https://github.com/getsince/test3/pull/481/files#diff-ecf0850f17c63f59b2690d88d79fbd691819ed852ee56937771756ddd24c19c8L60) is incorrect, I undestand now why you were confused  - I forgot to delete this comment when making some of the changes (sorry for this 🙏)

so actually "whatsapp" contact should be returned here

Reply:
since contact was shared later than timeslot -> was supposed to overwrite it

Reply:
I've updated the test and it's still wrong, the tests was successful before because of the order function clauses matched in the view.

Reply:
for some reason I thought that in this PR `interaction` is supposed (?) to determine the latest interaction mode => here the latest interaction mode would be `contact` then

nevertheless the current PR is probably safe to merge as well, since current client wouldn't allow for `timeslot` and `contact` to co-exist anyways (user would need to "cancel" contact / timeslot to send timeslot / contact)

so we can implement the above functionality (`interaction` being based on the latest interaction mode or rather one new interaction mode overriding content of the previous one -> guarantee of one interaction mode at a time for match) in a separate PR (probably in voicemail one)

Reply:
> since current client wouldn't allow for timeslot and contact to co-exist anyways (user would need to "cancel" contact / timeslot to send timeslot / contact)

but in the next version of the client I plan to remove this from client -> needs to be done in `voicemail` PR

Reply:
> for some reason I thought that in this PR interaction is supposed

no,  this pr refactors match_view db calls which were to calculate expiration

Reply:
no, I don't think it makes sense to do in voicemail PR

Reply:
moved to https://github.com/getsince/test3/issues/483